ext/liquid_c/variable.c

```diff
@@ -3,6 +3,137 @@
 #include "parser.h"
 #include <stdio.h>
 
+static ID id_rescue_strict_parse_syntax_error;
+
+static void compile_expression(vm_assembler_t *code, VALUE expression, bool const_expression)
+{
+    if (const_expression)
+        vm_assembler_add_push_const(code, expression);
+    else
+        vm_assembler_add_push_eval_expr(code, expression);
+}
+
+static int compile_each_keyword_arg(VALUE key, VALUE value, VALUE func_arg)
+{
+    vm_assembler_t *code = (vm_assembler_t *)func_arg;
+
+    vm_assembler_add_push_const(code, key);
+
+    bool is_const_expr = rb_respond_to(value, id_evaluate) == Qtrue;
```

Comment:
This seems backwards to me?

Reply:
Oops, this was always resulting in `is_const_expr = false`, because `rb_respond_to` doesn't return a VALUE, it returns a boolean `int` (i.e. 0 or 1).  It wasn't causing a test to fail because `context_evaluate` is used for non-constant expressions, which works with constants, just less efficiently.  I'll change this to `bool is_const_expr = !rb_respond_to(value, id_evaluate);` to fix it